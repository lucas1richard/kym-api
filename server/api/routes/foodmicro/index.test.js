const { expect, use } = require('chai');
const { jestSnapshotPlugin } = require('mocha-chai-jest-snapshot');

describe('foodmicro routes', () => {
  use(jestSnapshotPlugin());
  const routePrefix = '/api/foodmicro';

  describe(`GET ${routePrefix}/:id/v1`, () => {
    before(globals.seedTestData);
    after(globals.destroyAllHook);

    it('gets food with micronutrient data', async () => {
      const res = await globals.agent.getWithToken(`${routePrefix}/2514/v1`);
      expect(res.statusCode).equal(200);
      expect(res.body).toMatchSnapshot();
    });

    it('validates the :id param', async () => {
      const res = await globals.agent.getWithToken(`${routePrefix}/xyz/v1`);
      expect(res.statusCode).equal(400);
      expect(res.body).toMatchSnapshot();
    });
  });
});
