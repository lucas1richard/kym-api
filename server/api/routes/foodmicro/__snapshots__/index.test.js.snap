// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`foodmicro routes GET /api/foodmicro/:id/v1 gets food with micronutrient data 1`] = `
Object {
  "abbrevMicro": Object {
    "Alpha_Carot": "0.00",
    "Ash": "1.06",
    "Beta_Carot": "0.00",
    "Beta_Crypt": "0.00",
    "Calcium": "15.00",
    "Cholestrl": "85.00",
    "Choline_Tot": "85.30",
    "Copper": "0.05",
    "FA_Mono": "1.24",
    "FA_Poly": "0.77",
    "FA_Sat": "1.01",
    "Fiber": "0.00",
    "Folate_DFE": "4.00",
    "Folate_Tot": "4.00",
    "Folic_acid": "0.00",
    "Food_Folate": "4.00",
    "Iron": "1.04",
    "Lut_Zea": "0.00",
    "Lycopene": "0.00",
    "Magnesium": "29.00",
    "Manganese": "0.02",
    "Niacin": "13.71",
    "Panto_acid": "0.97",
    "Phosphorus": "228.00",
    "Potassium": "256.00",
    "Retinol": "6.00",
    "Riboflavin": "0.11",
    "Selenium": "27.60",
    "Sodium": "74.00",
    "Sugar": "0.00",
    "Thiamin": "0.07",
    "Vit_A_IU": "21.00",
    "Vit_A_RAE": "6.00",
    "Vit_B12": "0.34",
    "Vit_B6": "0.60",
    "Vit_C": "0.00",
    "Vit_D_IU": "5.00",
    "Vit_D_mcg": "0.10",
    "Vit_E": "0.27",
    "Vit_K": "0.30",
    "Water": "65.26",
    "Zinc": "1.00",
    "abbrev_id": 2514,
    "id": 2514,
  },
  "calories": "165",
  "carbohydrates": "0.00",
  "fat": "3.57",
  "foodDesc": Object {
    "CHO_Factor": "3.87",
    "ComName": "",
    "Fat_Factor": "9.02",
    "FdGrp_Cd": 500,
    "Long_Desc": "Chicken, broilers or fryers, breast, meat only, cooked, roasted",
    "ManufacName": "",
    "N_Factor": "6.25",
    "Pro_Factor": "4.27",
    "Ref_desc": "19% bone, 9% skin",
    "Refuse": 28,
    "SciName": "",
    "Short_Desc": "CHICKEN,BROILERS OR FRYERS,BRE",
    "Survey": "Y",
    "abbrev_id": 2514,
    "id": 1,
  },
  "gmwt_1": "140",
  "gmwt_2": "52",
  "gmwt_desc1": "1 cup, chopped or diced",
  "gmwt_desc2": "1 unit,  (yield from 1 lb ready-to-cook chicken)",
  "id": 2514,
  "main": "Chicken",
  "photo": null,
  "protein": "31.02",
  "sub": "broilers or fryers, breast, meat only, cooked, roasted",
  "user_uuid": "60f24265-ef45-4891-b4ef-e8fad4435308",
  "weights": Array [
    Object {
      "abbrev_id": 2514,
      "amount": "1.00",
      "description": "cup, chopped or diced",
      "gr_wgt": "140.00",
      "id": 1,
      "normalized": Object {
        "txt": "cup, chopped or diced (140 g)",
        "val": 1,
        "weight": 140,
      },
      "seq": 1,
    },
    Object {
      "abbrev_id": 2514,
      "amount": "1.00",
      "description": "unit (yield from 1 lb ready-to-cook chicken)",
      "gr_wgt": "52.00",
      "id": 2,
      "normalized": Object {
        "txt": "unit (yield from 1 lb ready-to-cook chicken) (52 g)",
        "val": 2,
        "weight": 52,
      },
      "seq": 2,
    },
    Object {
      "abbrev_id": 2514,
      "amount": "0.50",
      "description": "breast, bone and skin removed",
      "gr_wgt": "86.00",
      "id": 3,
      "normalized": Object {
        "txt": "breast, bone and skin removed (172 g)",
        "val": 3,
        "weight": 172,
      },
      "seq": 3,
    },
  ],
}
`;

exports[`foodmicro routes GET /api/foodmicro/:id/v1 validates the :id param 1`] = `
Array [
  Object {
    "context": Object {
      "label": "value",
    },
    "message": "ID_MUST_BE_NUMBER",
    "path": Array [],
    "type": "override",
  },
]
`;
